devops/jobs/UserRetirementArchiver.groovy

```diff
@@ -0,0 +1,80 @@
+/*
+
+Defines the job for archiving retired user statuses.
+
+Job config variables consumed for this job (via extraVars):
+  * ENVIRONMENT_DEPLOYMENT: The environment and deployment written as $environment-$deployment
+  * FOLDER_NAME: Name of jenkins "folder" under which to put this job.
+  * SECURE_GIT_CREDENTIALS: Name of jenkins git credential scoped to the UserRetirement "folder" in jenkins.
+  * ACCESS_CONTROL: List of github groups with read and build access to this job.
+  * ADMIN_ACCESS_CONTROL: List of github groups with full admin access to this job.
+  * CRON: Optional. Defines the cron line for scheduling the job.
+  * MAILING_LIST: Optional. A string containing a comma-delimited list of emails to notify when a build fails.
+
+This job expects the following credentials to be defined on the folder:
+  * tools-edx-jenkins-aws-credentials: file with key/secret in boto config format
+  * ${environment}-${deployment}-retirement-archive-upload-role: the role to aws sts assume-role
+
+*/
+package devops.jobs
+import static org.edx.jenkins.dsl.UserRetirementConstants.common_access_controls
+
+class UserRetirementArchiver {
+    public static def job = { dslFactory, extraVars ->
+
+        // Job config assertions:
+        assert (extraVars.get('MAILING_LIST', '') instanceof String) :
+            "Make sure MAILING_LIST is a single string (containing a comma-delimited list) and not a proper Yaml " +
+            "list or some other non-string type."
+
+        def job_name = extraVars.get("FOLDER_NAME") + '/archiver-' + extraVars.get('ENVIRONMENT_DEPLOYMENT')
+        dslFactory.job(job_name) {
```

Comment:
Note that i removed the `user-retirement-` prefix from the job name because now all user retirement jobs will be namespaced to the folder.  After expansion, this job should end up with a name like `User-Retirement/archiver-prod-edx`.